packages/cli/src/commonlib/m365LoginUserPassword.ts

```diff
@@ -0,0 +1,130 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+"use strict";
+
+import axios, { AxiosRequestConfig } from "axios";
+import dotenv from "dotenv";
+import qs from "querystring";
+
+import {
+  M365TokenProvider,
+  LogLevel,
+  TokenRequest,
+  Result,
+  FxError,
+  ok,
+  err,
+  LoginStatus,
+  UserError,
+} from "@microsoft/teamsfx-api";
+
+import * as cfg from "./common/userPasswordConfig";
+import CLILogProvider from "./log";
+import { ConvertTokenToJson, ErrorMessage } from "./codeFlowLogin";
+import { signedIn, signedOut } from "./common/constant";
+
+dotenv.config();
+
+const user = cfg.M365_ACCOUNT_NAME;
+const password = cfg.M365_ACCOUNT_PASSWORD;
+
+export class M365ProviderUserPassword implements M365TokenProvider {
+  private static instance: M365ProviderUserPassword;
+
+  private static accessToken: string | undefined;
+
+  public static getInstance(): M365ProviderUserPassword {
+    if (!M365ProviderUserPassword.instance) {
+      M365ProviderUserPassword.instance = new M365ProviderUserPassword();
+    }
+    return M365ProviderUserPassword.instance;
+  }
+
+  /**
+   * Get team access token
+   */
+  async getAccessToken(tokenRequest: TokenRequest): Promise<Result<string, FxError>> {
+    const data = qs.stringify({
+      client_id: cfg.client_id,
+      scope: tokenRequest.scopes,
+      username: user,
+      password: password,
+      grant_type: "password",
+    });
+
+    const config: AxiosRequestConfig = {
+      method: "post",
+      url: `https://login.microsoftonline.com/${
+        cfg.M365_TENANT_ID || "organizations"
+      }/oauth2/v2.0/token`,
+      headers: {
+        "Content-Type": "application/x-www-form-urlencoded",
+        Cookie:
+          "fpc=AmzaQu9yHbpLtMD2LmHazdRCGxwGAQAAAIW47NcOAAAA; x-ms-gateway-slice=estsfd; stsservicecookie=estsfd",
```

Comment:
what's this cookie for, hard-coded value?

Reply:
This has been removed, this does not have any effect on username password login.